tests/ui/mut_key.rs

```diff
@@ -24,18 +25,50 @@ impl Hash for Key {
     }
 }
 
+struct PtrToCell(*const Cell<usize>);
+#[derive(Eq, PartialEq)]
+struct MutPtr(*mut usize);
```

Comment:
Note that having a type that contains a field that is a pointer is different from having a type with a generic type parameter that may be a pointer type. `substs.type_at(0)` is the latter. So I'm not sure this test is meaningful.

Reply:
> Note that having a type that contains a field that is a pointer is different from having a type with a generic type parameter that may be a pointer type.

How come?  Surely they're going to generate exactly the same code, so should be treated equivalently by the compiler and this lint?

Reply:
They are different language constructs and they are represented differently when Clippy inspects them. The lint does not look at field types. And I don't think it should either. It would be expensive to recurse all fields, and we don't necessarily know which fields are hashed.

Reply:
> They are different language constructs

Ah OK that makes sense.  I still feel like it would be more logical to make them the same, but I'm sure there's a reason that the compiler folks have chosen this way of doing things.  In that case, it would make sense to remove these tests.

Reply:
Righto these test cases have been removed.